core/primitives/src/utils.rs

```diff
@@ -28,6 +30,13 @@ pub fn key_for_account(account_key: &AccountId) -> Vec<u8> {
     key_for_column_account_id(col::ACCOUNT, account_key)
 }
 
+pub fn key_for_data(account_id: &AccountId, data: &[u8]) -> Vec<u8> {
+    let mut bytes = key_for_account(account_id);
+    bytes.append(&mut ACCOUNT_DATA_SEPARATOR.to_vec());
+    bytes.append(&mut data.clone().to_vec());
```

Comment:
You don't need `append`:

> `fn append(&mut self, other: &mut Vec<T>)`
> Moves all the elements of other into Self, leaving other empty.

`.extend` or `.extend_from_slice` should work fine:

> `fn extend<I>(&mut self, iter: I)`
> `fn extend_from_slice(&mut self, other: &[T])`

```rust
    let mut bytes = key_for_account(account_id);
    bytes.extend(ACCOUNT_DATA_SEPARATOR);
    bytes.extend(data);
```